libraries/AP_IRLock/pixy_parser.cpp

```diff
@@ -0,0 +1,276 @@
+
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * pixy_parser.cpp
+ * *
+ */
+
+#ifdef PIXY_PARSER_USE_ASSERT
+#define PIXY_PARSER_ASSERT(x) {assert(x);}
+#else
+#define PIXY_PARSER_ASSERT(x) {}
+#endif
+
+#include "pixy_parser.h"
+
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+
+
+// Constructor - pixy_parser
+pixy_parser::pixy_parser() {
+    pixy_len = 0;
+    blob_buffer_write_idx = 0;
+    bytes_to_sof = 0;
+    bytes_to_block = 0;
+    blob_buffer[blob_buffer_write_idx].count = 0;
+    if_swap_buffer = 0;
+}
+
+// Destructor - pixy_parser
+pixy_parser::~pixy_parser() { }
+
+// Method - empty_pixyBuf
+void pixy_parser::empty_pixyBuf() {
+    for (size_t i=0; i<pixy_len; i++) {
+        pixy_buf[i] = 0;
+    }
+    pixy_len = 0;
+
+    // printf("\n--------------------------AFTER EMPTY: ");
+    for (size_t i=0; i<pixy_len; i++) {
+        // printf("%u, ", (unsigned)pixy_buf[i]);
+    }
+}
+
+// Method - print_buffer
+void pixy_parser::print_buffer() {
+    struct blob_buffer& writebuf = blob_buffer[blob_buffer_write_idx];
+    // printf("Buffer:");
```

Comment:
Remove all commented code or place in #ifdefs

Reply:
Done. Left print statements for testing